src/test/java/org/alfresco/repo/rawevents/EventBehaviorTest.java

```diff
@@ -0,0 +1,421 @@
+import org.alfresco.service.cmr.repository.StoreRef;
+import org.alfresco.service.namespace.QName;
+import org.alfresco.service.transaction.TransactionService;
+import org.alfresco.test_category.OwnJVMTestsCategory;
+import org.alfresco.util.ApplicationContextHelper;
+import org.alfresco.util.GUID;
+import org.alfresco.util.PropertyMap;
+import org.apache.camel.CamelContext;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.TestName;
+import org.springframework.context.ApplicationContext;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Provides a base set of tests for {@link EventBehaviour}
+ * 
+ * @author Cristian Turlica
+ */
+@Category(OwnJVMTestsCategory.class)
+public class EventBehaviorTest
+{
+    private static final String TEST_NAMESPACE = "http://www.alfresco.org/test/EventBehaviorTest";
+
+    private ApplicationContext ctx;
+    private TransactionService transactionService;
+    private ContentService contentService;
+    private PolicyComponent policyComponent;
+    private NodeService nodeService;
+    private AuthenticationComponent authenticationComponent;
+    private UserTransaction txn;
+    private NodeRef rootNodeRef;
+    private NodeRef contentNodeRef;
+
+    private CamelContext camelContext;
+    private EventProducer eventProducer;
+    private ObjectMapper messagingObjectMapper;
+
+    private boolean policyFired = false;
+    private boolean newContent = true;
+
+    @Rule
+    public TestName name = new TestName();
+
+    @Before
+    public void setUp() throws Exception
+    {
+        ctx = ApplicationContextHelper.getApplicationContext();
+        transactionService = (TransactionService) ctx.getBean("TransactionService");
+        nodeService = (NodeService) ctx.getBean("NodeService");
+        contentService = (ContentService) ctx.getBean(ServiceRegistry.CONTENT_SERVICE.getLocalName());
+        policyComponent = (PolicyComponent) ctx.getBean("policyComponent");
+        authenticationComponent = (AuthenticationComponent) ctx.getBean("authenticationComponent");
+
+        camelContext = (CamelContext) ctx.getBean("alfrescoCamelContext");
+        eventProducer = (EventProducer) ctx.getBean("alfrescoEventProducer");
+        messagingObjectMapper = (ObjectMapper) ctx.getBean("messagingObjectMapper");
+
+        // authenticate
+        authenticationComponent.setSystemUserAsCurrentUser();
+    }
+
+    @After
+    public void tearDown() throws Exception
+    {
+        try
+        {
+            authenticationComponent.clearCurrentSecurityContext();
+        }
+        catch (Throwable e)
+        {
+            // ignore
+        }
+    }
+
+    private UserTransaction getUserTransaction()
```

Comment:
Please use `@Transactional` instead for cleaner code. It rollbacks by default.
If required RetryingTransactionHelper is a better choice as some of the tests might intermittently fail without retries.

Reply:
We can't use @Transaction for now because we need OwnJVMTestsCategory... I used RetryingTransactionHelper instead.